### ISSUE TITLE
GoProController and GoProControllerUI installation issues

### ISSUE THREAD

**Issue by pjotawake** [joshvillbrandt/GoProController#7]
Lately I have been fiddling around with the GoPro software. The GoPro Hero library installs fine and seems to be working on my Ubuntu 14.4 (64bit) machine.

However, after installing the GoProController I run into problems - the page at http://localhost/ shows a no permission error

"Forbidden - You don't have permission to access / on this server"

Looking at the Apache configuration I can see that access is granted for /home/GoProControllerUI/_public. But this directory does not exist.

Apparently, it should be there after installing GoProControllerUI. However, this installation runs into the error of a ambiguous "node" binary which prevents correct installation (see below).

---

npm http GET https://registry.npmjs.org/lru-cache/-/lru-cache-2.7.0.tgz
/bin/sh: 1: node: not found
gyp: Call to 'node -e "require('nan')"' returned exit status 127. while trying to load binding.gyp
npm http GET https://registry.npmjs.org/sigmund/-/sigmund-1.0.1.tgz
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/node-gyp/lib/configure.js:431:16)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:797:12)
gyp ERR! System Linux 3.19.0-31-generic
gyp ERR! command "nodejs" "/usr/bin/node-gyp" "rebuild"
gyp ERR! cwd /home/gopro/GoProControllerUI/node_modules/auto-reload-brunch/node_modules/ws/node_modules/bufferutil
gyp ERR! node -v v0.10.25
gyp ERR! node-gyp -v v0.10.10
gyp ERR! not ok 
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian

npm http 200 https://registry.npmjs.org/lru-cache/-/lru-cache-2.7.0.tgz

npm http 200 https://registry.npmjs.org/is-property
npm http GET https://registry.npmjs.org/is-property/-/is-property-1.0.2.tgz
npm http 200 https://registry.npmjs.org/is-property/-/is-property-1.0.2.tgz

> GoProControllerUI@0.2.4 postinstall /home/gopro/GoProControllerUI
> node_modules/bower/bin/bower install --config.interactive=false

/usr/bin/env: node: No such file or directory
npm ERR! weird error 127
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian

**Comment by joshvillbrandt** [comment]
The error seems pretty straight forward.

```
/bin/sh: 1: node: not found
```

Seems like you need to install Node. I recommend trying to install Node v0.10 or v0.12 using [nvm](https://github.com/creationix/nvm).

**Comment by pjotawake** [comment]
Thanks for your reply - I did try to use the default packages from Ubuntu repository, but unfortunately the node version was too old (0.10.24). And apparently they have a name conflict with some other package using the filename 'node', therefore they renamed the node binary into something else.

So I now installed the latest v0.10 node which is 0.10.40 using NVM, and I can get through the compilation process, though I do see some errors about a "failed optional dep" in the GoProControllerUI package (see below).

Also the webinterface works now.

Thanks again!

gopro@zotac:~/GoProControllerUI$ npm install

> GoProControllerUI@0.2.4 preinstall /home/gopro/GoProControllerUI
> rm -rf node_modules bower_components
> 
> utf-8-validate@1.1.0 install /home/gopro/GoProControllerUI/node_modules/auto-reload-brunch/node_modules/ws/node_modules/utf-8-validate
> node-gyp rebuild

Usage: gyp_main.py [options ...] [build_file ...]

gyp_main.py: error: no such option: --no-parallel
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onCpExit (/home/gopro/.nvm/v0.10.40/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:343:16)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Linux 3.19.0-31-generic
gyp ERR! command "node" "/home/gopro/.nvm/v0.10.40/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/gopro/GoProControllerUI/node_modules/auto-reload-brunch/node_modules/ws/node_modules/utf-8-validate
gyp ERR! node -v v0.10.40
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 
npm WARN optional dep failed, continuing fsevents@0.3.8
npm WARN optional dep failed, continuing utf-8-validate@1.1.0
